bittide/src/Bittide/ProcessingElement/Util.hs

```diff
@@ -22,25 +22,45 @@ import qualified Data.ByteString as BS
 import qualified Data.IntMap as I
 import qualified Data.List as L
 
+-- | Given a `Maybe Int` and a list, if the `Maybe Int` is `Just s`, pad the list to
+-- size `s` with the given element. If the length of the list is greater than `s`,
+-- throw an error. If the `Maybe Int` is `Nothing`, return the list as is.
+padToSize :: HasCallStack => Maybe Int -> a -> [a] -> [a]
+padToSize Nothing _ l = l
+padToSize (Just s) a xs
+  | s >= l = xs <> L.replicate (s - l) a
+  | otherwise = error "Bittide.ProcessingElement.Util: List is longer than the specified size"
```

Comment:
I guess since this `error` is only hit during TemplateHaskell execution it's easy enough to figure out what exactly is the issue? Mostly concerned about ease of debugging when you encounter this

Reply:
Would you like me to extend the error message with the relevant sizes?

Reply:
Since this is only ever run in TemplateHaskell, you could make this function return `Q [a]` and then use `iListPadded <- padToSize "instruction memory" (fmap ((`div` 4) . (+3)) iSize) 0 iList` (and the same for the data memory).

With the name as an argument the error message can be better, and if it's in the `Q` monad you also get to use `reportError`, so if both sizes are wrong both errors will be reported.